### PR TITLE
Implemented Improved Auto-Mapping Tool

### PR DESCRIPTION
**Pull request by robellgabriel**
new features:
- reduced nodes in neighbourhood graph to only be placed at intersection 
- added func `add_dest_node()` which adds an arbitrary destination node to graph by finding the closest edge on the graph and connecting its nodes with the dest node, from there dijkstra simply finds shortest path. If there is no edge near by (meaning dest node outside of node graph), it simply connects dest node to the closest node on the map
- added func `optimize_path()` which currently generates points between intersections in the given path from dijkstra

WIP:
- add code in TESTGUI_helper.py to `optimize_path()` to generate curve at intersection
-->this was more tricky than expected as the angle at which the car approaches intersection is almost unique for each
- run tests on new auto-mapping tool
- clean-up mapping subsystem folder (ie: remove unused files, organize code, etc)

### CODE REVIEW

**Review by phucla0122 — APPROVED** [review]

File: src/mapping_navigation/scripts/TestGUI.py [hunk 1]
```diff
@@ -360,58 +290,116 @@
...
+        path = find_shortest_path(NH_G, NH_startpoint, dest_node)
+    elif current_image == CITY:
+        add_dest_node(C_G, dest_node, C_pointList)
+        path = find_shortest_path(C_G, C_startpoint, dest_node)
 
     #create line on canvas showing generated path
-    draw_line(path)
+    better_path = optimize_path(path)
+    draw_line(better_path)
+    print(path)
```
Comment (nicolastuttle) [inline src/mapping_navigation/scripts/TestGUI.py:387]: This line is cluttering the terminal when we run the tool

File: src/mapping_navigation/scripts/TestGUI_helper.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import math
```
Comment (nicolastuttle) [inline src/mapping_navigation/scripts/TestGUI_helper.py:1]: What is this script used for?

File: src/mapping_navigation/scripts/TestGUI.py (hunk 1 above)
Comment (robellgabriel) [reply src/mapping_navigation/scripts/TestGUI.py:387]: Yeah it was used for testing purposes, will delete all the print statements in my next PR once i finished implementing the curve

File: src/mapping_navigation/scripts/TestGUI_helper.py (hunk 2 above)
Comment (robellgabriel) [reply src/mapping_navigation/scripts/TestGUI_helper.py:1]: Was trying out the curve implementation based off my design in the report, when you run it plots the points which shows a curve. Currently just used to test out my theory and will also delete this script in my nextPR once I finish implementing the curve.

**Review by nicolastuttle — APPROVED** [review]
I like the new path finding. Makes paths a lot more flexible!

**Pull request merged**